pages/products/[id].jsx

```diff
@@ -0,0 +1,18 @@
+import { useRouter } from 'next/router';
+import useSwr from 'swr';
+import React from 'react';
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
```

Comment:
`fetcher` method is duplicated in a few components. Move it into a separate file and import where needed